src/_pytest/assertion/rewrite.py

```diff
@@ -330,16 +328,16 @@ def _write_pyc(state, co, source_stat, pyc):
             fp.write(struct.pack("<LL", mtime, size))
             fp.write(marshal.dumps(co))
     except EnvironmentError as e:
-        state.trace("error writing pyc file at %s: errno=%s" % (pyc, e.errno))
+        state.trace("error writing pyc file at {}: errno={}".format(pyc, e.errno))
         # we ignore any failure to write the cache file
         # there are many reasons, permission-denied, __pycache__ being a
         # file etc.
         return False
     return True
 
 
-RN = "\r\n".encode("utf-8")
```

Comment:
Not sure this is correct... doesn't `encode` use the default system locale?

Reply:
in python3+ it uses UTF-8 if no encoding is specified: https://github.com/asottile/pyupgrade/issues/138